packages/runtimes/hmr/src/HMRRuntime.js

```diff
@@ -41,7 +41,9 @@ export default (new Runtime({
           port != null &&
             // Default to the HTTP port in the browser, only override
             // in watch mode or if hmr port != serve port
-            (!options.serveOptions || options.serveOptions.port !== port)
+            (!options.serveOptions ||
+              options.serveOptions.port !== port ||
+              forcePortOverride)
```

Comment:
If the dev server and HMR server are on the same port, won't there be an error that the port is already in use? How does this option work?

Reply:
This was allowing the logic on https://github.com/parcel-bundler/parcel/blob/045662f5839450c60f6a757961d601e7500bbb3a/packages/reporters/dev-server/src/ServerReporter.js#L45 to be forced through, which seemed to fix the configuration in Jira.

I've followed up with a new change which I believe is the root cause, where the server was using HTTPS but the client wasn't.